test/SlowTests/Server/Documents/PeriodicBackup/FtpBasicTests.cs

```diff
@@ -0,0 +1,375 @@
+                try
+                {
+                    await ftpServerHost.StartAsync(CancellationToken.None);
+                    var settings = new FtpSettings { Url = "ftp://127.0.0.1:21/internal", UserName = "anonymous", Password = "itay@ravendb.net" };
+                    using (var client = new RavenFtpClient(settings))
+                    using (var store = GetDocumentStore())
+                    {
+                        var config = Backup.CreateBackupConfiguration(ftpSettings: settings, name: "ftpBackupTest",
+                            retentionPolicy: new RetentionPolicy { MinimumBackupAgeToKeep = TimeSpan.FromSeconds(15) });
+                        var backupId = (await store.Maintenance.SendAsync(new UpdatePeriodicBackupOperation(config))).TaskId;
+                        for (int i = 0; i < 3; i++)
+                        {
+                            using (var session = store.OpenSession())
+                            {
+                                session.Store(new User { Name = "itay" });
+                                session.SaveChanges();
+                            }
+
+                            var lastEtag = store.Maintenance.Send(new GetStatisticsOperation()).LastDocEtag;
+                            await Backup.RunBackupAndReturnStatusAsync(Server, backupId, store, expectedEtag: lastEtag, timeout: 120000);
+                        }
+                        await Task.Delay(TimeSpan.FromSeconds(15) + TimeSpan.FromSeconds(3));
+                        using (var session = store.OpenAsyncSession())
+                        {
+                            await session.StoreAsync(new User { Name = "itay" });
+                            await session.SaveChangesAsync();
+                        }
+                        var etag = store.Maintenance.Send(new GetStatisticsOperation()).LastDocEtag;
+                        await Backup.RunBackupAndReturnStatusAsync(Server, backupId, store, isFullBackup: true, expectedEtag: etag, timeout: 120000);
+                        var folders = client.GetFolders();
+                        var foundFolders = 0;
+                        for (int i = 0; i < folders.Count; i++)
+                        {
+                            var isExist = folders[i].Contains("CanUploadBackupsWithDeletion");
+                            if (isExist)
+                                foundFolders++;
+                        }
+                        Assert.Equal(1, foundFolders);
+                    }
+                }
+                finally
+                {
+                    BackupConfigurationHelper.SkipMinimumBackupAgeToKeepValidation = false;
+                    await ftpServerHost.StopAsync(CancellationToken.None);
+                }
+            }
+        }
+
+        [FtpFact]
+        public async Task CanUploadFileOnEncrypted()
+        {
+            CertificateUtils.CreateCertificateAuthorityCertificate("auth", out var caKey, out var caName);
```

Comment:
Wouldn't it be better to use Certificates.GenerateAndSaveSelfSignedCertificate method? this way you are using cached one instead of generating it from scratch each time (which takes time)

Reply:
Done